exception Invalid {
1: optional string message
}

exception BadRequest {
1: optional string message
}
exception UserNotAllowed {
1: optional string message
}

exception Precondition {
1: optional string message
}

exception Range {
1: optional string message
}

exception Unauthenticated {
1: optional string message
}

exception Permission {
1: optional string message
}

exception NotFound {
1: optional string message
}
exception DoesNotExist {
1: optional string message
}

exception Aborted {
1: optional string message
}

exception Exists {
1: optional string message
}
exception Exhausted {
1: optional string message
}

exception Cancelled {
1: optional string message
}

exception DataLoss {
1: optional string message
}

exception UnknownException {
1: optional string message
}
exception UnknownError {
1: optional string message
}
exception InternalError {
1: optional string message
}
exception InternalException {
1: optional string message
}
exception ServiceError {
1: optional string message
}

exception ServiceException {
1: optional string message
}
exception ServerError {
1: optional string message
}
exception ServerException {
1: optional string message
}

exception Unimplemented {
1: optional string message
}
exception Unavailable {
1: optional string message
}
exception DeadlineExceeded {
1: optional string message
}
